components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/internal/service/impl/ActionManagementServiceImpl.java

```diff
@@ -156,8 +157,10 @@ public Action updateAction(String actionType, String actionId, Action action, St
         ActionDTO updatingActionDTO = buildActionDTO(resolvedActionType, actionId, action);
 
         DAO_FACADE.updateAction(updatingActionDTO, existingActionDTO, IdentityTenantUtil.getTenantId(tenantDomain));
-        auditLogger.printAuditLog(ActionManagementAuditLogger.Operation.UPDATE, updatingActionDTO);
-        return getActionByActionId(actionType, actionId, tenantDomain);
+        Action updatedAction = getActionByActionId(actionType, actionId, tenantDomain);
+        ActionDTO auditActionDTO = buildAuditActionDTO(updatedAction);
+        auditLogger.printAuditLog(ActionManagementAuditLogger.Operation.UPDATE, auditActionDTO);
```

Comment:
Is this correct ?
Since this a PATCH operation, we only need to log the updating attributes.
With this change I hope we are changing that behaviour.